zebra-rpc/src/methods.rs

```diff
@@ -526,6 +549,52 @@ where
         }
         .boxed()
     }
+
+    fn get_address_tx_ids(
+        &self,
+        addresses: Vec<String>,
+        start: u32,
+        end: u32,
+    ) -> BoxFuture<Result<Vec<String>>> {
+        let mut state = self.state.clone();
+        let mut response_transactions = vec![];
+        let start = Height(start);
+        let end = Height(end);
+
+        async move {
+            if start > end {
+                return Err(Error::invalid_params(
+                    "End height should be at least start height",
+                ));
+            }
+
+            for address in addresses {
+                let valid_address: Address = address.parse().map_err(|_| {
+                    Error::invalid_params(format!("Provided address is not valid: {}", address))
+                })?;
+
+                let request =
+                    zebra_state::ReadRequest::TransactionsByAddresses(valid_address, start, end);
```

Comment:
I am passing start and end to the request, unsure what to call from there yet but i hope this extra arguments make sense. I mean, will be good to have just the transactions in the range rather that all the history for the address. Hope this will be possible.

Reply:
The state functions will look something like this:
https://github.com/ZcashFoundation/zebra/pull/4038/commits/bcd8d0b8c9b85bb44eafe417b32c2eb4f200383e#diff-6a151e25fa9d83c40553b33ee7b4f0789ec8d836824b55392b165b228f0de533R176-R182

We'll need to add the height filters to the function in a later PR, because it will make it much faster.

Reply:
Ok, good to know this, thanks!